administrator/components/com_content/models/article.php

```diff
@@ -706,6 +705,53 @@ public function featured($pks, $value = 0)
 	}
 
 	/**
+	 * Method to store the token generated.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since 3.7
+	 */
+	private function shareTokenGenerate()
+	{
+		jimport('joomla.user.helper');
+		$token = JUserHelper::genRandomPassword(16);
+
+		return $token;
+	}
+
+	/**
+	 * Method to store the token generated.
+	 *
+	 * @param   string  $title  The title of the shared article.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since   3.7
+	 */
+	public function shareToken($title)
+	{
+		$table = $this->getTable('share', 'ContentTable');
+		$token = $this->shareTokenGenerate();
+
+		try
+		{
+			$db = $this->getDbo();
+			$query = $db->getQuery(true);
+
+			$data = array('articleId' => $this->get('id'), 'title' => $title, 'sharetoken' => $token);
+
+			$db->setQuery($query);
+			$table->save($data);
+
+			return $db->execute();
```

Comment:
Imo don't return the boolean here.

I would return the token, e.g. return $token;